### ISSUE TITLE
Standalone `$` is accepted as a token in declarative macros.

### ISSUE THREAD

**Issue by cafce25** [rust-lang/rust#153142]
I tried this code:

```rust
macro_rules! foo {
    ($) => {}
}
```

I expected to see this happen: an error because a standalone `$` doesn't match any of the [grammar rules from the reference](https://doc.rust-lang.org/reference/macros-by-example.html#r-macro.decl.syntax)

Instead, this happened: the code compiles, you can successfully invoke the macro like this `foo!($);`

[Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2024&gist=2a936093bcf3e6aabac04c42c6ff85e7)

### Meta
Latest stable on the playground (1.93.1)
Also on the nightly verson:
```
Nightly version: 1.95.0-nightly
(2026-02-24 859951e3c7c9d0322c39)
```
</p>
</details>
